api/v1/helpers/verbs/heartbeatUtils.js

```diff
@@ -102,28 +102,26 @@ function getKey(collectorName, changeType) {
  * collector.
  *
  * @param {Object} generator - The Generator that has been changed
- * @param {Array} oldCollectors - The names of the collectors that were associated
- *  with this generator before the update.
- * @param {Array} newCollectors - The names of the collectors that are associated
- *  with this generator now, after the update has been applied.
+ * @param {Array} oldCollector - The name of the Collector this Generator was
+ *  assigned to before the update.
+ * @param {Array} newCollector - The name of the Collector this Generator is
+ *  assigned to after the update.
  */
-function trackGeneratorChanges(generator, oldCollectors, newCollectors) {
-  const genId = generator.id;
-  const allCollectors = Array.from(new Set([...oldCollectors, ...newCollectors]));
-
-  Promise.all(allCollectors.map(name => getChangedIds(name)))
-  .then((collectorChanges) => {
-    let promises = [];
-    collectorChanges.forEach((changedIds, i) => {
-      const collectorName = allCollectors[i];
+function trackGeneratorChanges(generator, oldCollector, newCollector) {
+  if (oldCollector === newCollector) {
+    return trackChangesForCollector(oldCollector, UPDATED, generator);
+  } else {
+    return Promise.all([
+      trackChangesForCollector(oldCollector, DELETED, generator),
+      trackChangesForCollector(newCollector, ADDED, generator),
+    ]);
+  }
 
-      // determine the change type
-      let change;
-      const inOld = oldCollectors.includes(collectorName);
-      const inNew = newCollectors.includes(collectorName);
-      if (inOld && !inNew) change = DELETED;
-      if (!inOld && inNew) change = ADDED;
-      if (inOld && inNew) change = UPDATED;
+  function trackChangesForCollector(collectorName, change, generator) {
```

Comment:
Defining trackChangesForCollector function inside trackGeneratorChanges function is resulting in multiple eslint errors on my sublime - I would like to know if there is an advantage in doing so.

Reply:
Personally I much prefer using internal functions in a case like this where it isn't used anywhere else and makes no sense outside the context of the calling function. It keeps it self-contained and makes it easier to understand.

Reply:
You already approved so I will go ahead and merge if there are no objections.

Reply:
Please do. For some reason, squash and merge is not enabled for me right now.